src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -279,6 +283,23 @@ export const getTopNavConfig = (
       testId: 'shareTopNavButton',
       run: (anchorElement) => {
         if (share && !embeddableId) {
+          const currentState = stateContainer.getState();
+          const searchParams = parse(history.location.search);
+          const params: VisualizeLocatorParams = {
+            visId: savedVis?.id,
+            filters: currentState.filters,
+            refreshInterval: undefined,
+            timeRange: data.query.timefilter.timefilter.getTime(),
+            type: currentState.vis.type,
+            uiState: currentState.uiState,
+            query: currentState.query,
+            vis: currentState.vis,
```

Comment:
Would the Visualize app be able to query the saved objects service for this object using the `visId`?

Just curious for now. It could be something to optimize in the future.

Reply:
That's a really good point 👍🏻  I'll open an issue for tracking that after we merge this PR.